DataDog/serverless-sample-app
# Feat/sst v2 sample

### What does this PR do?

This implements a sample of deploying the existing stacks via [sst.dev v2](https://docs.sst.dev/what-is-sst).

### Motivation

The infrastructure configuration is similar to AWS CDK but the development flow is different enough to warrant a sample. One example is when you run `sst dev`, i.e. you are running it from your laptop, in which case you want to set your Datadog env (`DD_ENV`) to your local `stage` (`SST_STAGE`).  

I have used SST for a few apps, so I thought I'd include an SST sample in this repo. 

Linked to [issue 23](https://github.com/DataDog/serverless-sample-app/issues/23).

### Testing Guidelines

Assuming you've got AWS credentials set up and you have a Datadog secret ARN set in your environment (`DD_SECRET_ARN`), and you have installed the node modules, you should be able to run:

`npm run dev:sst`

This will deploy the stack with the stage `personal` (unless you change it to anything else). You should then be able to use the API Gateway URL by setting the environment variable `API_ENDPOINT` to run the tests:

`npm run test -- product-service`

### Additional Notes

This is an implementation of SST v2, not v3 (named "Ion"). 

### Types of Changes

- [ ✔️] New feature

### Check all that apply
